### PR TITLE
Closes #61 | Create dataset loader for mc4-indo #61

### LINKED ISSUE #61: Create dataset loader for mc4-indo
Dataloader name: `mc4_indo/mc4_indo.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?mc4_indo

|     Dataset| mc4_indo |
|-------------|---|
| Description | A thoroughly cleaned version of the Indonesia split of the multilingual colossal, cleaned version of Common Crawl's web crawl corpus (mC4). This portion represents the Indonesian language content that has been extracted and processed from the larger mC4 dataset. The extraction and cleaning process was conducted by AllenAI and resulted in a curated collection of Indonesian language data. For more information about the original mC4 dataset and its preparation, please refer to the source hosted at the address https://huggingface.co/datasets/allenai/c4. |
| Subsets | - |
| Languages | ind |
| Tasks | Language Modeling |
| License | Open Data Commons License Attribution family (odc-by) |
| Homepage | https://huggingface.co/datasets/indonesian-nlp/mc4-id |
| HF URL | https://huggingface.co/datasets/indonesian-nlp/mc4-id |
| Paper URL | https://aclanthology.org/2021.naacl-main.41 |

### PR DESCRIPTION
**Pull request by williamnixon20**
Please name your PR after the issue it closes. You can use the following line: "Closes #ISSUE-NUMBER" where you replace the ISSUE-NUMBER with the one corresponding to your dataset.

Closes #61 

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/mc4_indo/mc4_indo.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+
+_DATASETNAME = "mc4_indo"
+_DESCRIPTION = """\
+    A thoroughly cleaned version of the Indonesia split of the multilingual colossal, cleaned version of Common Crawl's web crawl corpus (mC4). This portion represents the Indonesian language content that has been extracted and processed from the larger mC4 dataset. The extraction and cleaning process was conducted by AllenAI and resulted in a curated collection of Indonesian language data. For more information about the original mC4 dataset and its preparation, please refer to the source hosted at the address https://huggingface.co/datasets/allenai/c4.
+"""
+
+_HOMEPAGE = "https://huggingface.co/datasets/indonesian-nlp/mc4-id"
+_LICENSE = Licenses.ODC_BY.value
+_LOCAL = False
```
Comment (jamesjaya) [inline seacrowd/sea_datasets/mc4_indo/mc4_indo.py:22]: remove unused _LOCAL

File: seacrowd/sea_datasets/mc4_indo/mc4_indo.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+_LANGUAGES = ["ceb"]
+_DATASETNAME = "cebuaner"
```
Comment (jamesjaya) [inline seacrowd/sea_datasets/mc4_indo/mc4_indo.py:24]: Wrong _LANGUAGES and _DATASETNAME, please fix

File: seacrowd/sea_datasets/mc4_indo/mc4_indo.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        id_ = 0
+        for filepath in filepaths:
+            with gzip.open(open(filepath, "rb"), "rt", encoding="utf-8") as f:
+                for line in f:
+                    if line:
+                        example = json.loads(line)
+                        if self.config.schema == "source":
+                            sourcee_json = {"text": str(example["text"]), "timestamp": str(example["timestamp"]), "url": str(example["url"])}
+
+                            yield id_, sourcee_json
```
Comment (jamesjaya) [inline seacrowd/sea_datasets/mc4_indo/mc4_indo.py:139]:
```suggestion
                            yield id_, example
```

Comment (jamesjaya) [reply seacrowd/sea_datasets/mc4_indo/mc4_indo.py:139]: Maybe we can just load the example?

**Review by jamesjaya — CHANGES_REQUESTED** [review]

File: seacrowd/sea_datasets/mc4_indo/mc4_indo.py (hunk 2 above)
Comment (williamnixon20) [reply seacrowd/sea_datasets/mc4_indo/mc4_indo.py]: so sorry, completely missed this :(

File: seacrowd/sea_datasets/mc4_indo/mc4_indo.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+_DATASETNAME = "mc4_indo"
+_DESCRIPTION = """\
+    A thoroughly cleaned version of the Indonesia split of the multilingual colossal, cleaned version of Common Crawl's web crawl corpus (mC4). This portion represents the Indonesian language content that has been extracted and processed from the larger mC4 dataset. The extraction and cleaning process was conducted by AllenAI and resulted in a curated collection of Indonesian language data. For more information about the original mC4 dataset and its preparation, please refer to the source hosted at the address https://huggingface.co/datasets/allenai/c4.
+"""
+
+_HOMEPAGE = "https://huggingface.co/datasets/indonesian-nlp/mc4-id"
+_LICENSE = Licenses.ODC_BY.value
+
+_LANGUAGES = ["ind"]
+_DATASETNAME = "mc4_indo"
```
Comment (ryanignatius) [inline seacrowd/sea_datasets/mc4_indo/mc4_indo.py:24]: `_DATASETNAME` already exists in line 15,
we can remove it here

**Review by ryanignatius — APPROVED** [review]

**Review by jamesjaya — APPROVED** [review]

**Pull request merged**